src/WebJobs.Script/Description/DotNet/Compilation/CSharp/CSharpCompilation.cs

```diff
@@ -100,10 +100,18 @@ private static bool IsOrUsesAssemblyType(ITypeSymbol typeSymbol, IAssemblySymbol
                 using (var assemblyStream = new MemoryStream())
                 using (var pdbStream = new MemoryStream())
                 {
-                    var compilationWithAnalyzers = _compilation.WithAnalyzers(GetAnalyzers());
-                    var diagnostics = await compilationWithAnalyzers.GetAnalyzerDiagnosticsAsync();
+                    Compilation compilation = _compilation;
+                    var diagnostics = ImmutableArray<Diagnostic>.Empty;
+
+                    if (compilation.Options.OptimizationLevel == OptimizationLevel.Debug)
+                    {
+                        var compilationWithAnalyzers = compilation.WithAnalyzers(GetAnalyzers());
```

Comment:
What about the WithAnalyzers call in GetDiagnostics()? When is that path exercised?

Reply:
That is only exercised in active development scenarios (on non-emit paths), so we want to keep that information.